xxxx-release-and-versioning.md

```diff
@@ -0,0 +1,87 @@
+# OIP-xxxx: eWallet Release & Versioning
+
+## Abstract
+
+This document specifies a procedure for releasing and versioning to allow for stable production environment. OmiseGO eWallet Server (hereafter "eWallet Server") is meant to be deployed by a various entities, which may range from a personal store to an enterprise (hereafter "Provider") and thus require a stable operation and smooth upgrade path that maximizes compatibilities.
+
+## Overview
+
+The eWallet release is consist of a stable release and a development release. A stable release is meant to be run by a Provider that require stability and compatibility, while a development release is meant for development and testing propose. The stable releases should be versioned using [SemVer](https://semver.org/).
+
+## Specifications
+
+### Stable Release
+
+A stable release is a release that provides compatibility and stability guarantee and must pass an extensive end-to-end testing and migration testing from supported versions prior to release.
+
+| Test subject        | Required |
+| ------------------- | -------- |
+| Linting             | **Yes**  |
+| Unit testing        | **Yes**  |
+| Integration testing | **Yes**  |
+| End to end testing  | **Yes**  |
+| Migration testing   | **Yes**  |
+
+A stable release will be provided as a Docker image based on Alpine Linux. The Docker image will be tagged with the release version (e.g. `v1.0.3`) and will use a `:latest` tag to refer to the latest stable release.
+
+Only the latest `MAJOR.MINOR.PATCH` version of the last 2 `MAJOR.MINOR` versions will be supported. However, we may provide security fixes to non-supported versions as the team see fit. For demostration purpose, given that we released the following versions at a specific date, our support condition shall be:
+
+| Version | Release Date | Support                                    |
+| ------- | ------------ | ------------------------------------------ |
+| 1.2.2   | 2018-06-12   | **Supported**, latest `1.2`                |
+| 1.2.1   | 2018-04-01   | Not supported, not latest `1.2`            |
+| 1.2.0   | 2018-01-01   | Not supported, not latest `1.2`            |
+| 1.1.1   | 2017-06-20   | **Supported**, latest `1.1`                |
+| 1.1.0   | 2017-06-20   | Not supported, not latest `1.1`            |
+| 1.0.2   | 2018-02-01   | Not supported, `1.0` discontinued (hotfix) |
+| 1.0.1   | 2017-08-01   | Not supported, `1.0` discontinued          |
+| 1.0.0   | 2016-08-01   | Not supported, `1.0` discontinued          |
+
+In the figure above, version `1.0.2` is released as a security fix outside of the support period. This release is considered a hotfix release. However, even though version `1.0.2` is released after the supported version `1.1.1`, the version `1.0.2` won't be guarantee to receive any further updates. Provider is encouraged to upgrade to latest supported version.
```

Comment:
I associate "hotfix" with fixes made to production environment (and somewhat quick and dirty), probably because of my past background. Should we clarify the definition a bit that:

> A hotfix is a patch that that addresses one or more specific issues on a version that is no longer supported.

?

Reply:
@T-Dnzt what term should we use? My thought is that "hotfix" means something that goes directly to release regardless whether they're still supported.

Reply:
I have the same understanding as you @sirn, I think `hotfix` is alright. Not limited to non-supported releases, maybe adding one more line to the table with a `hotfix` pushed on a supported release will clarify things.